components/plurals/src/lib.rs

```diff
@@ -875,3 +875,97 @@ where
             .unwrap_or(end)
     }
 }
+
+#[derive(Debug)]
+#[non_exhaustive]
+/// A bag of values for different plural cases.
+pub struct PluralElements<'a, T: ?Sized> {
+    /// The value used for the [`PluralCategory::Zero`] case
+    pub zero: Option<&'a T>,
+    /// The value used for the [`PluralCategory::One`] case
+    pub one: Option<&'a T>,
+    /// The value used for the [`PluralCategory::Two`] case
+    pub two: Option<&'a T>,
+    /// The value used for the [`PluralCategory::Few`] case
+    pub few: Option<&'a T>,
+    /// The value used for the [`PluralCategory::Many`] case
+    pub many: Option<&'a T>,
+    /// The value used for the [`PluralCategory::Other`] case
+    pub other: &'a T,
+    /// The value used when the [`PluralOperands`] are exactly 0.
+    pub explicit_zero: Option<&'a T>,
+    /// The value used when the [`PluralOperands`] are exactly 1.
+    pub explicit_one: Option<&'a T>,
+}
+
+impl<'a, T: ?Sized + PartialEq> PluralElements<'a, T> {
+    /// Creates a new [`PluralElements`] with the given default value.
+    pub fn new(other: &'a T) -> Self {
+        Self {
+            other,
+            zero: None,
+            one: None,
+            two: None,
+            few: None,
+            many: None,
+            explicit_zero: None,
+            explicit_one: None,
+        }
+    }
+
+    /// Sets the value for [`PluralCategory::Zero`].
+    pub fn with_zero_value(self, zero: Option<&'a T>) -> Self {
+        Self {
+            zero: zero.filter(|&t| t != self.other),
+            ..self
+        }
+    }
+
+    /// Sets the value for [`PluralCategory::One`].
+    pub fn with_one_value(self, one: Option<&'a T>) -> Self {
+        Self {
+            one: one.filter(|&t| t != self.other),
+            ..self
+        }
+    }
```

Comment:
These functions consume `self` and return `Self`, is it only change the field that we need to change or it creates a `Self` object each time ?

Reply:
These will inline. `self -> Self` is required for this to be usable in a builder chain.